src/tools/pkcs11-tool.c

```diff
@@ -3378,7 +3393,25 @@ gen_key(CK_SLOT_ID slot, CK_SESSION_HANDLE session, CK_OBJECT_HANDLE *hSecretKey
 			FILL_ATTR(keyTemplate[n_attr], CKA_KEY_TYPE, &key_type, sizeof(key_type));
 			n_attr++;
 		}
-		else {
+
+		else if (strncasecmp(type, "HKDF:", strlen("HKDF:")) == 0) {
```

Comment:
HKDF: seems to be defined in different places. May be a const char* for it would allow easy changing it

Reply:
The code I added for HKDF is a copy/paste of the GENERIC case.
We would have to do the same for the other key types otherwise the code would not be consistent.
I will address your remark in a new PR.